src/openforms/submissions/migrations/0005_temporaryfileupload_legacy_and_more.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 4.2.11 on 2024-06-26 14:24
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("submissions", "0004_auto_20231128_1536"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="temporaryfileupload",
+            name="legacy",
+            field=models.BooleanField(
+                default=False,
+                help_text="Whether the instance is linked to a submission instance.",
+                verbose_name="legacy",
+            ),
+        ),
+        migrations.AddField(
+            model_name="temporaryfileupload",
+            name="submission",
+            field=models.ForeignKey(
+                help_text="Submission the temporary file upload belongs to.",
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
+                to="submissions.submission",
+                verbose_name="submission",
+            ),
+        ),
+        migrations.AddConstraint(
+            model_name="temporaryfileupload",
+            constraint=models.CheckConstraint(
+                check=models.Q(
+                    models.Q(("legacy", False), ("submission__isnull", False)),
+                    models.Q(("legacy", True), ("submission__isnull", True)),
+                    _connector="OR",
+                ),
+                name="non_legacy_submission_not_null",
+            ),
+        ),
```

Comment:
so:

1. legacy is added, default is `False` and all `submission` columns are `NULL`
2. the constraint is added and says that an upload needs to either
     * have legacy `False` and submission `NOT NULL`
     * have legacy `True` and submission `NULL`

that will crash during migration when legacy upload records exist - I think this migration needs to be incorporated in the migration tests too to prevent this from happening

The constraint can only be added after the data is fixed through a data migration